src/FlashtestationRegistry.sol

```diff
@@ -79,9 +80,16 @@ contract FlashtestationRegistry is
      * @notice The TEE must be registered with a quote whose validity is verified by the attestationContract
      * @dev In order to mitigate DoS attacks, the quote must be less than 20KB
      * @dev This is a costly operation (5 million gas) and should be used sparingly.
+     * @dev appData should be versioned as it is likely to change, and it should be in part app-agnostic (version and pubkey) and in part app-specific (remainer of the bytes). Changes to the app-specific part of app
      * @param rawQuote The raw quote from the TEE device. Must be a V4 TDX quote
+     * @param appData The extended attested to data
      */
-    function registerTEEService(bytes calldata rawQuote) external limitBytesSize(rawQuote) nonReentrant {
+    function registerTEEService(bytes calldata rawQuote, bytes calldata appData)
```

Comment:
readability: this comment should explain why and how we use the app data hash to bind the TEE-attested reportData to the appData. Otherwise it's confusing why this data isn't just included in the rawQuote and why the cryptographi binding is necessary.